src/DynamoPackages/PackageLoader.cs

```diff
@@ -418,16 +418,45 @@ public Package ScanPackageDirectory(string directory, bool checkCertificates)
                 // prevent loading unsigned packages if the certificates are required on package dlls
                 if (checkCertificates)
                 {
-                    CheckPackageNodeLibraryCertificates(directory, discoveredPkg);
+                    CheckPackageNodeLibraryCertificates(directory, discoveredPackage);
                 }
 
-                // prevent duplicates
-                if (LocalPackages.All(pkg => pkg.Name != discoveredPkg.Name))
+                var discoveredVersion = CheckAndGetPackageVersion(discoveredPackage.VersionName, discoveredPackage.Name, discoveredPackage.RootDirectory);
+
+                var existingPackage = LocalPackages.FirstOrDefault(package => package.Name == discoveredPackage.Name);
+
+                // Is this a new package?
+                if (existingPackage == null)
                 {
-                    Add(discoveredPkg);
-                    return discoveredPkg; // success
+                    // Yes
+                    Add(discoveredPackage);
+                    return discoveredPackage; // success
                 }
-                throw new LibraryLoadFailedException(directory, String.Format(Properties.Resources.DulicatedPackage, discoveredPkg.Name, discoveredPkg.RootDirectory));
+
+                var existingVersion = CheckAndGetPackageVersion(existingPackage.VersionName, existingPackage.Name, existingPackage.RootDirectory);
+
+                // Is this a duplicated package?
+                if (existingVersion == discoveredVersion)
+                {
+                    // Duplicated with the same version
+                    throw new LibraryLoadFailedException(directory, String.Format(Properties.Resources.DulicatedPackage, discoveredPackage.Name, discoveredPackage.RootDirectory));
+                } 
+                
+                // Is the existing version newer?
+                if (existingVersion > discoveredVersion)
+                {
+                    // Newer version already exist
+                    Log(String.Format(Properties.Resources.DuplicatedOlderPackage, discoveredPackage.Name, discoveredPackage.RootDirectory, existingVersion.ToString(), discoveredVersion.ToString()), WarningLevel.Moderate);
+                    return null;
+                }
+
+                // Older version exist, replace with newer version
+                Remove(existingPackage);
```

Comment:
if `existingPackage` is already installed/loaded then this won't do anything during this session - right?

Reply:
Correct.

Reply:
This happens when you add/remove package directories. I think we should do something about it but it feels like it should be a separate task.

Reply:
okay, so this workflow only works well during first startup of Dynamo after this PR? Thats okay with me as long as we file a followup

Reply:
I will file something.

Reply:
This was actually easier to fix than I expected. I have added the case when a new version of a package is found but the old version is already loaded. This will show up as an notification for now.

Reply:
LGTM

Reply:
@pinzart90 It happens when everything reloads after adding a new package directory.